src/libraries/System.Formats.Tar/tests/TarWriter/TarWriter.WriteEntryAsync.Tests.cs

```diff
@@ -322,5 +322,62 @@ await using (TarReader reader = new TarReader(archiveStream))
                 }
             }
         }
+
+        public static IEnumerable<object[]> WriteEntry_TooLongName_Throws_Async_TheoryData()
+            => TarWriter_WriteEntry_Tests.WriteEntry_TooLongName_Throws_TheoryData();
+
+        [Theory]
+        [MemberData(nameof(WriteEntry_TooLongName_Throws_Async_TheoryData))]
+        public async Task WriteEntry_TooLongName_Throws_Async(TarEntryFormat entryFormat, TarEntryType entryType, string name)
+        {
+            using TarWriter writer = new(new MemoryStream());
```

Comment:
Super nit: I've been trying to make sure we also test IAsyncDisposable in async test methods:
```suggestion
            await using TarWriter writer = new(new MemoryStream());
```